src/qibosoq/rfsoc_server.py

```diff
@@ -42,26 +46,37 @@ def execute_program(self, data: dict) -> dict:
         Returns:
             (dict): dictionary with two keys (i, q) to lists of values
         """
-        if data["operation_code"] == "execute_pulse_sequence":
-            program = ExecutePulseSequence(global_soc, data["cfg"], data["sequence"], data["qubits"])
-        elif data["operation_code"] == "execute_single_sweep":
-            program = ExecuteSingleSweep(global_soc, data["cfg"], data["sequence"], data["qubits"], data["sweeper"])
+        opcode = OperationCode(data["operation_code"])
+        args = ()
+        if opcode is OperationCode.EXECUTE_PULSE_SEQUENCE:
+            programcls = ExecutePulseSequence
+        elif opcode is OperationCode.EXECUTE_SWEEPS:
+            programcls = ExecuteSweeps
+            args = (Sweeper(**sweeper) for sweeper in data["sweepers"])
```

Comment:
Pay attention to this: this is not a tuple, but a generator (essentially an inline syntax to define a function that yields).

I like it a lot, but might become counter-intuitive.

In particular, being essentially a function, it is a closure: it captures variables from the environment it is defined in, but it uses it only when it is iterated.

```py
>>> l = list(range(5))
>>> f = (i for i in l)
>>> l.append(10)
>>> list(f)
[0, 1, 2, 3, 4, 10]
```

If you want a tuple instead, just iterate it immediately:
```suggestion
            args = tuple(Sweeper(**sweeper) for sweeper in data["sweepers"])
```